File: resources/scop_1.5.1/examples/event_source.cpp
// event_source.cpp - DMI - 7-9-02

/* Usage: event_source [ <source> ]   (default source is "news") */

#include <scop.h>

int main(int argc, char **argv)
{
	int sock;
	int count = 1;
	char msg[80];
	
	sock = scop_open("localhost", "event_source");
	scop_set_source_hint(sock, argc > 1 ? argv[1] : (char *)"news");
	
	while(1)
	{
		sprintf(msg, "Item %d", count);
		scop_emit(sock, msg);
		count++;
		sleep(1);
	}
	
	return 0;
}

File: resources/scop_1.5.1/java/examples/EventSource.java
// EventSource.java - DMI - 21-9-02

/* Usage: java EventSource [ <source> ]   (default source is "news") */

import java.io.*;

public class EventSource
{
	public static void main(String[] args)
	{
		SCOP scop;
		int count = 1;

		scop = new SCOP("localhost", "EventSource");
		scop.set_source_hint(args.length > 0 ? args[0] : "news");

		while(true)
		{
			scop.emit("Item " + count);
			count++;
			try { Thread.sleep(1000); } catch(Exception e) {}
		}
	}
}
